panels/spoolman.py

```diff
@@ -0,0 +1,405 @@
+                ' 63.27-34.925 0-63.27-28.345-63.27-63.27 0-34.925 28.345-63.27 63.27-63.27z"/></g></svg>').encode())
+            loader.close()
+            self._icon = loader.get_pixbuf()
+        #            self._icon = Gtk.Image.new_from_pixbuf(pixbuf)
+
+        return self._icon
+
+    icon = property(get_icon)
+    name = property(get_name)
+
+
+class Panel(ScreenPanel):
+    apiClient: KlippyRest
+    _active_spool_id: int = None
+
+    def spool_compare_func(self, model, row1, row2, user_data):
+        spool1 = model.get_value(row1, 0)
+        spool2 = model.get_value(row2, 0)
+        return spool1.id - spool2.id
+
+    def _on_material_filter_changed(self, sender):
+        treeiter = sender.get_active_iter()
+        model = sender.get_model()
+        self._filters["material"] = model[treeiter][0]
+        self._filterable.refilter()
+
+    def __init__(self, screen, title):
+        super().__init__(screen, title)
+        self.apiClient = screen.apiclient
+        self.timeFormat = '%Y-%m-%d %H:%M' if self._config.get_main_config().getboolean("24htime",
+                                                                                        True) else '%Y-%m-%d %I:%M %p'
+
+        GObject.type_register(SpoolmanSpool)
+        self._filters = {}
+        self._model = Gtk.TreeStore(SpoolmanSpool.__gtype__)
+        self._materials = Gtk.ListStore(str, str)
+
+        self._filterable = self._model.filter_new()
+        self._filterable.set_visible_func(self._filter_spools)
+
+        sortable = Gtk.TreeModelSort(self._filterable)
+        sortable.set_sort_func(0, self.spool_compare_func, None)
+        sortable.set_sort_column_id(0, Gtk.SortType.ASCENDING)
+
+        self.scroll = self._gtk.ScrolledWindow()
+
+        sbox = Gtk.Box(spacing=0)
+        sbox.set_vexpand(False)
+
+        clear_active_spool = self._gtk.Button("cancel", _("Clear Spool"), "color2", self.bts, Gtk.PositionType.LEFT, 1)
+        clear_active_spool.get_style_context().add_class("buttons_slim")
+        clear_active_spool.connect('clicked', self.clear_active_spool)
+
+        refresh = self._gtk.Button("refresh", _("Refresh"), "color1", self.bts, Gtk.PositionType.LEFT, 1)
+        refresh.get_style_context().add_class("buttons_slim")
+        refresh.connect('clicked', self.load_spools)
+
+        sort_btn = self._gtk.Button("arrow-up", _("ID"), "color4", self.bts, Gtk.PositionType.RIGHT, 1)
+        sort_btn.connect("clicked", self.change_sort)
+        sort_btn.get_style_context().add_class("buttons_slim")
+
+        switch = Gtk.Switch()
+        switch.set_hexpand(False)
+        switch.set_vexpand(False)
+        switch.set_active(self._config.get_config().getboolean("spoolman", "hide_archived", fallback=True))
+        switch.connect("notify::active", self.switch_config_option, "spoolman", "hide_archived", self.load_spools)
+
+        name = Gtk.Label()
+        name.set_markup(f"Hide archived")
```

Comment:
you can do:
`name = Gtk.Label(label=_("Hide archived"))`
but an f-string without {} is not valid